### DISCUSSION TITLE
Retrieve only specific values due to performance

### DISCUSSION THREAD

**Discussion by eliasu** [statamic/cms#10434] [Q&A]
Hi, 

I have a collection named **formats**.
Each **format** contains a lot (200) of events – entries from another collection.

 if I try to retrieve the title of the format like so
```antlers
{{ format.title }}
```
It is very slow. I think because it looks at all those events and their properties as well ( i checked this with the debugbar and saw, that the performance increases if i delete some of that event entries within the format entry).

Is it possible to adjust the query, so that it will only give me back the **title** without the need for "scanning" the whole format entry with all its events?

Thank you.

**Comment by edalzell** [comment]
Is this in production? If so do you have Stache Watcher on?

**Comment by eliasu** [comment]
No this is not in production … Stache Watcher was on. If I turn it Off the speed increases, but, its still very slow. It takes 8 s to render the name of the format entry :(

Any further ideas?

**Reply by edalzell** [reply]
you're in a collection loop?

**Reply by eliasu** [reply]
No. i'm on the ```show.antlers.html``` template of a collection named artists. 
Every artist has multiple events (concerts). 
Each events has one single format (a festival for example). 

So imagine it like: an artist can have multiple concerts at different festivals. 
So I'm doing something like this:

```antlers
{{#  Start: /artists/show.antlers.html #}}

{{# looping through all events of this artist (2-3 items) #}}
{{ events scope="event" }}
    
    {{# get the title of the event's format – this line is weirdly slow #}}
    {{ event.format.title }}
{{ /events }}
```

So this is a part of the markdown of an artist
```md
---
id: ce44dda2-ed5b-4a14-879c-3e6444f03511
blueprint: artist
title: testartist
events:
  - 67c4f44c-f059-4af2-8238-f9dfe23c49f8
  - b5910b13-db1d-4615-8885-ee665d752217
```

This is a part of the markdown of the corresponding event. As you see an event has an artist field and a format field. 
```md
---
id: 67c4f44c-f059-4af2-8238-f9dfe23c49f8
blueprint: event
title: 'Test Title Concert'
artists:
  - ce44dda2-ed5b-4a14-879c-3e6444f03511
format: 8fcb7d06-11dc-46db-867b-db4b75e7f21a
```

And last but not least the markdown of the format. As you can see, in the "many_events" fiel of this format, there are many events as well as the event from above. 
```md
---
id: 8fcb7d06-11dc-46db-867b-db4b75e7f21a
blueprint: format
title: 'Test Title Festival'
many_events:
  - 67c4f44c-f059-4af2-8238-f9dfe23c49f8
  - 84a793f5-3be3-4a13-9503-fb213a47f2e4
  - 2c2da4d5-b96d-41c7-b2b0-e1d0d8498601
  - 8e3dce4f-94ee-48a2-9384-594989cb0ce6
  ... (many more)
```

I hope that helps to understand.
